rust-toolchain

```diff
@@ -1 +0,0 @@
-nightly-2019-11-05
```

Comment:
The benefit of having this file is that you don't have to set up the toolchain manually yourself (from a clean checkout):

```
rob@artemis:~/src/rust-hypervisor-firmware (josephlr-cleanup)$ cargo xbuild --release --target target.json
error: The sysroot can't be built for the Stable channel. Switch to nightly.
note: run with `RUST_BACKTRACE=1` for a backtrace
```

It could just be `nightly` but then it might be out of sync with Travis and i've also had problems with the nightly tree builds failing and also missing useful tools (like clippy, rustfmt or rls) so I would prefer to keep this file and keep it in sync with the .travis.yml (annoying, I know.)

As for testing nightly I think it might be possible to create a travis test matrix so that it tests with nightly as well as the pinned version.

Reply:
Makes sense, I updated the pinned toolchain to use today's. As for CI, I have some ideas to make it better, but that's best left for  a different PR